File: backend/app/schemas/audit.py
from datetime import datetime
from pydantic import BaseModel
from app.schemas.common import Meta


class AuditLogOut(BaseModel):
    id: int
    company_id: int
    user_id: int | None
    entity_type: str
    entity_id: int
    action: str
    before: dict | None
    after: dict | None
    request_id: str | None
    ip: str | None
    created_at: datetime

    class Config:
        from_attributes = True


class AuditLogList(BaseModel):
    data: list[AuditLogOut]
    meta: Meta

File: gestion-materiel-frontend/src/lib/api/audit.ts
import { http } from "@/lib/api/http";
import type { Meta } from "@/lib/api/assets";

export type AuditLogOut = {
  id: number;
  company_id: number;
  user_id: number | null;
  entity_type: string;
  entity_id: number;
  action: string;
  before: Record<string, any> | null;
  after: Record<string, any> | null;
  request_id: string | null;
  ip: string | null;
  created_at: string;
};

export type AuditLogList = {
  data: AuditLogOut[];
  meta: Meta;
};

export type ListAuditLogsParams = {
  entity_type?: string;
  entity_id?: number;
  user_id?: number;
  action?: string;
  limit?: number;
  offset?: number;
};

export async function listAuditLogs(params: ListAuditLogsParams = {}) {
  const res = await http.get<AuditLogList>("/admin/audit-logs", { params });
  return res.data;
}
